adaptive/learner/balancing_learner.py

```diff
@@ -107,10 +110,13 @@ def strategy(self, strategy):
             self._ask_and_tell = self._ask_and_tell_based_on_loss
         elif strategy == "npoints":
             self._ask_and_tell = self._ask_and_tell_based_on_npoints
+        elif strategy == "cycle":
+            self._ask_and_tell = self._ask_and_tell_based_on_cycle
+            self._cycle = itertools.cycle(range(len(self.learners)))
```

Comment:
Now the cycle will be reset every time the strategy is set dynamically. I'm not sure what the best thing to do here is.

Reply:
That might be the intention? We could also just put in in `__init__`...